src/gpu/vulkan/SDL_gpu_vulkan.c

```diff
@@ -1010,83 +759,76 @@ typedef struct VulkanOcclusionQuery
 
 /* Cache structures */
 
-/* Descriptor Set Layout Caches*/
-
-#define NUM_DESCRIPTOR_SET_LAYOUT_BUCKETS 1031
-
-typedef struct DescriptorSetLayoutHash
-{
-    VkDescriptorType descriptorType;
-    Uint32 bindingCount;
-    VkShaderStageFlagBits stageFlag;
-} DescriptorSetLayoutHash;
+/* MurmurHash3 */
 
-typedef struct DescriptorSetLayoutHashMap
+/* Finalization mix - force all bits of a hash block to avalanche */
+inline static Uint64
+murmur3_fmix32(Uint64 h)
 {
-    DescriptorSetLayoutHash key;
-    VkDescriptorSetLayout value;
-} DescriptorSetLayoutHashMap;
+	h ^= h >> 16;
+	h *= 0x85ebca6b;
+	h ^= h >> 13;
+	h *= 0xc2b2ae35;
+	h ^= h >> 16;
 
-typedef struct DescriptorSetLayoutHashArray
-{
-    DescriptorSetLayoutHashMap *elements;
-    Sint32 count;
-    Sint32 capacity;
-} DescriptorSetLayoutHashArray;
+	return h;
+}
 
-typedef struct DescriptorSetLayoutHashTable
-{
-    DescriptorSetLayoutHashArray buckets[NUM_DESCRIPTOR_SET_LAYOUT_BUCKETS];
-} DescriptorSetLayoutHashTable;
+/* The following function uses code from the .NET Runtime
+ * MIT License (MIT)
```

Comment:
We'll have to replace this to keep everything as zlib, but hopefully it won't be hard to do that - I feel like we should have this already in SDL somewhere

Reply:
If necessary I can try to ask someone to get the OK to relicense this as zlib for you